projects/com.oracle.truffle.llvm.nodes.impl/src/com/oracle/truffle/llvm/nodes/impl/base/LLVMTerminatorNode.java

```diff
@@ -29,25 +29,46 @@
  */
 package com.oracle.truffle.llvm.nodes.impl.base;
 
-import com.oracle.truffle.api.CompilerDirectives.CompilationFinal;
 import com.oracle.truffle.api.frame.VirtualFrame;
-import com.oracle.truffle.llvm.nodes.base.LLVMNode;
+import com.oracle.truffle.api.instrumentation.Instrumentable;
+import com.oracle.truffle.api.source.SourceSection;
+import com.oracle.truffle.llvm.nodes.base.LLVMStatementNode;
 
 /**
  * This node represents a terminator instruction in LLVM IR. This node decides which basic block is
  * executed next, or terminates execution of a function.
  *
+ * Since this class has an additional field, it can't be simply wrapped by the wrapper node
+ * generated from super class {@link LLVMStatementNode}. Therefore this class is also {@link Instrumentable}.
+ *
  * @see <a href="http://llvm.org/docs/LangRef.html#terminator-instructions">terminator
  *      instructions</a>
  */
-public abstract class LLVMTerminatorNode extends LLVMNode {
+@Instrumentable(factory = LLVMTerminatorNodeWrapper.class)
+public abstract class LLVMTerminatorNode extends LLVMStatementNode {
 
-    @CompilationFinal private final int[] successors;
```

Comment:
I thought that the CompilationFinal annotation is used for fields which should be treated as final by Truffle during runtime while the field actually is not marked final. Since the field is marked final I thought the annotation would be obsolete, but correct me if I am wrong. I did the same thing for the source section field in LLVMStatementNode class.